spec/workers/purge_worker_spec.rb

```diff
@@ -0,0 +1,92 @@
+require "rails_helper"
+
+RSpec.describe PurgeWorker, type: :worker do
+  describe ".perform_async" do
+    subject(:perform_async) { described_class.perform_async }
+
+    it "enqueues 1 job with expected options" do
+      expect { perform_async }
+        .to change(described_class, :jobs)
+          .from([])
+          .to(
+              [
+                hash_including(
+                  "retry" => true,
+                  "queue" => "default",
+                  "args" => [],
+                  "class" => "PurgeWorker"
+                )
+              ]
+          )
+    end
+  end
+
+  describe "#perform" do
+    subject(:perform) { described_class.new.perform }
+
+    let(:bulk_submission) { create(:bulk_submission, :with_original_file, :with_result_file, expires_at:) }
+    let(:submission) do
+ create(:submission, first_name: 'Rosie', last_name: 'Conway', nino: 'JC654321A', dob: '1977-03-08'.to_date,  
+bulk_submission:) end
+    let(:expires_at) { Time.zone.now }
+
+    before do
+      bulk_submission
+      submission
+    end
+
+    it_behaves_like "applcation worker logger"
+
+    context "when the bulk submission has expired" do
+      it "purges the original file" do
+        perform
+        bulk_submission.reload
+        expect(bulk_submission.original_file.attachment).to be_nil
+        expect(bulk_submission.original_file.blob).to be_nil
```

Comment:
**nit** can use [compound expectations](https://makandracards.com/makandra/58102-rspec-3-allows-chaining-multiple-expectations) here

Reply:
Have amended to use have_attributes to make one line expectation